### ISSUE TITLE
Eight small fixes: messages, exit code, CONTRIBUTING drift, `.gitignore` write

### ISSUE THREAD

**Issue by vlad-ryzhkov** [zernie/agent-lint#176]
Tested on `vigiles@21.0.1` (npm) = `681edf7`.

Eight small things found while adopting vigiles in a 51-skill monorepo. None deserves its own issue;
together they are under twenty lines of change. Split them off if any one is worth its own thread.

---

### 1. `skill-resource-resolves` should name `sharedDirs` in the message

A `SKILL.md` mentions `scripts/lint.sh` in prose — a file at the repo root, and it exists. vigiles
resolves it relative to the skill dir (`src/cli.ts:4511`):

```
⚠ my-skill: bundled resource "scripts/lint.sh" (line 106) is referenced
  but missing — the agent reads the instruction and gets nothing.
```

With `"skill-resource-resolves": "error"` this reddens CI over a non-problem.

The fix exists and works: `"sharedDirs": ["scripts", "docs", "tests"]`. It is documented only in
`docs/skills-monorepo.md`, and the message says nothing about it. I found it by reading the docs
end to end; a CI log has no hint.

**Patch:** append half a line to the message — "if it resolves from the repo root, add its dir to
`sharedDirs` in `.vigilesrc.json`". Removes the main false-positive source in a skills monorepo, no
new code.

---

### 2. No way to force the single-report view on a repo with many bundles

At the root of a repo with 29 bundles, `vigiles audit .` switches to leaderboard mode. There is no
way to get the full ring report for the root. The flag list:

```
--capability-diff --catalog-only --concurrency --fail-on-widen --harness --json
--max-rules --md --min-prompts --model --no-html --no-interactive --no-json
--no-open --no-serve --out --prompts --serve --trials --yes
```

So `audit` has to be called per directory — 29 runs for 29 bundles, which is how our weekly CI job
ended up written.

**Patch:** a `--single` (or `--no-leaderboard`) flag that parses the given directory as one harness
regardless of how many manifests are inside. The mode-selection branch already exists; the flag just
pins it.

---

### 3. `--out` is a silent no-op in leaderboard mode

```
$ npx vigiles audit . --no-interactive --no-open --out=report
EXIT=0
$ ls report
ls: report: No such file or directory
```

The same flag on a single bundle writes `vigiles-report.html` and `vigiles-report.json`. A silent
no-op is how a CI job ships an empty artifact and stays green — that is exactly how I found it.

**Patch:** one line — in leaderboard mode with `--out` given, print
`⚠ --out is ignored in leaderboard mode (no per-bundle report is produced)`. Or, if that reads
better, write the leaderboard itself to `--out` as JSON.

---

### 4. `STABILITY.md` names a removed verb and the wrong major

1. `STABILITY.md:3` — "vigiles is at **v12**". Published is `21.0.1`, latest tag `v21.0.1`.
2. `STABILITY.md:14-16` lists `scan` among the **stable** CLI verbs: "`init`, `compile`, `lint`,
   `test`, `eval`, `scan`, `generate`". But:

```
$ npx vigiles scan .
Unknown command: "scan"
```

`grep '"scan"' src/cli.ts` returns nothing.

The second one is the heavier of the two: a stability contract naming a removed verb. And `vigiles
lint` cannot catch it — CLI verb references in markdown are not marked up in any way.

**Patch:** drop `scan` from the list, fix the major on line 3.

Separate thought, not part of that patch: a ref kind that checks a mentioned CLI verb exists — the
same mechanism `cmd()` uses for npm scripts. That would catch the class rather than the instance.

---

### 5. Unknown command exits 1 where the documented contract says 2

`docs/cli.md:50-56` fixes the contract: `1` is "ran, and what you asked about is bad", `2` is "could
not do what you asked — unknown flag, unknown harness, nothing to audit".

```
$ npx vigiles lint --bogus   → EXIT=2
$ npx vigiles bogus          → EXIT=1
```

An unknown command is "could not do what you asked", so it should be 2. A script that tells "found
problems" from "could not start" by exit code reads a typo'd command name as the former.

**Patch:** `src/cli.ts:5962` — `process.exit(1)` → `process.exit(2)` in the "Unknown command" branch,
matching the unknown-flag path (`src/cli-flag-check.ts:209`).

---

### 6. `CONTRIBUTING.md` "Adding a new validation rule" points at files that no longer exist

Tried to follow `CONTRIBUTING.md` while preparing these patches. Steps 1-4 of that section are not
executable:

- step 1 — "Add the rule name to `RulesConfig` in `src/types.ts`" (`CONTRIBUTING.md:94`) → no such
  file;
- step 2 — "Add the default to `RULE_PACKS` in `src/validate.ts`" (`:95`) → no such file;
- step 3 — "Implement the check inside the `validate()` function" → no such function;
- step 4 — "Add tests in `src/validate.test.ts`" → no such file.

Three more claims in the same file:

- `:90` and `:127` — "Keep the single-file core architecture — `validate.ts` contains all validation
  logic". `src/` currently has 197 top-level `.ts` files (385 including subdirectories), plus
  `src/core/`.
- `:52` — "Tests use Node.js built-in test runner (`node:test`). No extra test framework needed".
  `package.json:75` is `"test": "npm run build && vitest run"`, and the package exports a `jest`
  adapter at `:53`.
- `:70` — "Run locally: `npx vigiles CLAUDE.md`" → `Unknown command: "CLAUDE.md"`.

The "Adding a new linter" section right below is accurate and useful: `LINTERS` in
`src/core/linters.ts`, `BUILTIN_LINTERS` in `src/core/spec.ts`, all present.

**Patch:** delete the "Adding a new validation rule" section or rewrite it after the "Adding a new
linter" model, and fix the three claims above. Deleting is cheaper than maintaining.

---

### 7. The zero-fire diagnostic knows the plugin id it just created, but does not print it

Credit first: on a zero-fire result `measureTriggerRate` does not report 0% as a fact — it prints an
ordered list of likely causes. That saved me a rerun worth roughly $10; I went and read the code
instead of guessing. Worth keeping and worth copying into other measurements.

One improvement. Cause 1 says (`src/eval.ts:2842-2843`):

```
1. the id in `fired` — `skillResolved` matches the NAMESPACED id (`<plugin>:<skill>`);
   a bare name silently never matches;
```

But with `skillsDir` the plugin name is not the user's choice: `src/eval.ts:2445` hardcodes
`vigiles-loose-skills`. At that point the runtime knows the exact id that should have matched, and
does not print it.

**Patch:** substitute it into the message:

```
1. the id in `fired` — with skillsDir your skills install as
   `vigiles-loose-skills:<name>`, e.g. `vigiles-loose-skills:mysql-designer`;
   a bare name silently never matches;
```

A format string, no new code. Cause 1 turns from "go check yourself" into "copy this".

---

### 8. `--out` outside the repo appends a `../../../…` entry to `.gitignore`

`audit` adds its reports to `.gitignore` — stated behaviour (`docs/cli.md`: "both auto-added to
.gitignore"). The path is written relative to the repo root with no check that it is inside it. With
`--out` pointing at `/tmp`:

```diff
+# vigiles audit report (generated)
+../../../../../../../private/tmp/.../vigrep/vigiles-report.json
+../../../../../../../private/tmp/.../vigrep/vigiles-report.html
```

That entry ignores nothing — the file is outside the tree and `.gitignore` does not reach it. Each
new `--out` path adds its own block, so a script that runs `audit` over bundles into separate
directories fills `.gitignore` with junk. Found in a CI job that walks 5 bundles.

Separately: a command the docs call a read-only report quietly edits `.gitignore`. For `--out`
inside the repo that is expected and documented; for a path outside it, it is a file edit with no
upside.

**Patch:** in `ensureReportGitignored` (`src/cli.ts:7320`), skip the write when `--out` resolves
outside the root — if `relative(repoRoot, outDir)` starts with `..`, leave `.gitignore` alone. One
condition where the block is currently appended unconditionally.

**Comment by zernie** [comment]
All eight shipped in #186 (`94af23a5`). Every one reproduced exactly as written; nothing here needed re-diagnosing. Point by point, with the two that turned out to be more than cosmetic called out.

| # | what shipped |
|---|---|
| 1 | `skill-resource-resolves` names the fix: "If it resolves from the repo root instead, add its directory to `sharedDirs` in .vigilesrc.json" |
| 2 | `--single` — audit a many-bundle root as one harness |
| 3 | `--out` in leaderboard mode prints `⚠ --out is ignored here …` instead of silently writing nothing |
| 4 | `STABILITY.md` — `scan` → `audit`, v12 → v21 |
| 5 | unknown command exits **2** |
| 6 | `CONTRIBUTING.md` — the dead section rewritten |
| 7 | the zero-fire diagnostic prints the resolved namespace |
| 8 | `--out` outside the repo no longer touches `.gitignore` |

**#8 was the one that mattered most, and your framing undersold it.** You filed it as junk accumulating in a file. It is also a straight violation of a contract this repo writes down for itself — `audit` is documented as a read-only report, and the "not evil" rule in `CLAUDE.md` opens with *a READ never writes*. Writing an unreachable `../../../../private/tmp/…` entry into a tracked file is that rule broken, for zero benefit, and the fix is the one-line containment check you proposed. Tested in both directions: an `--out` inside the repo is still gitignored, because disabling the useful half would be the opposite defect.

**#4 got a mechanical floor, not just a correction.** You noted `vigiles lint` cannot catch it, and you are right — `self-command-refs` is deliberately high-precision and only inspects a `vigiles <cmd>` reference in a *command context*, while STABILITY.md lists its verbs as a prose list of backticked words. Widening the detector is how `doc-refs` ended up disabled. So the one page where the cost is highest got a test instead (`src/stability-verbs.test.ts`), asserting **both** directions against the real `VERBS` list: a promised verb that does not exist, and a shipped verb the contract forgets. A mutation reintroducing `scan` fails it.

Your separate thought — a ref kind that checks a mentioned CLI verb, like `cmd()` does for npm scripts — is the class-level fix and is not in this PR. It is worth its own issue if you want to file it; the honest reason it did not ship here is that its false-positive surface (every prose mention of the word "test" or "audit") is exactly the shape that got `doc-refs` turned off, and calibrating it is not a same-PR job.

**#7 shipped slightly wider than you proposed.** You suggested substituting the hardcoded `vigiles-loose-skills` into cause 1. The runtime resolves the namespace for the `pluginDir` path too, so it is now carried on the report (`TriggerRateReport.namespace`) and the message names whichever one actually applies. Cause 1 turns from "go check yourself" into "copy this" for both install paths, not just `skillsDir`.

**#6, on your point that the section was not executable at any step:** confirmed — `src/types.ts`, `src/validate.ts`, `validate()` and `src/validate.test.ts` are all gone, and `src/` holds 197 top-level files against a claimed "single-file core". Rewritten on the "Adding a new linter" model you correctly identified as the accurate one, and every file the new section names is asserted to exist.

**#5:** fixed at the branch you pointed at; the unknown-flag path already exited 2, and a test now asserts the two agree.

Thank you for the exit codes, the line numbers, and the reproductions — all eight were actionable without a single clarifying question, which is rare.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by zernie** [comment]
Follow-up on **item 8** only — the guard you proposed is now gone, which is worth a note since it is not what you suggested.

You proposed one condition in `ensureReportGitignored`: if `relative(repoRoot, outDir)` starts with `..`, skip the write. That is correct and it shipped that way in #186.

In #188 (`7693c71`) it was **deleted** and replaced with a type:

```ts
export type RepoRelativePath = string & { readonly __repoRelative: unique symbol };
export function repoRelative(root: string, target: string): RepoRelativePath | null;
```

`repoRelative` returns `null` for anything absolute or escaping, and the `.gitignore` writer accepts nothing else. The reason for moving it off the write site: a guard has to be repeated at every write site and the next one gets written without it — the same shape as #173, where four sibling compilers each needed the guard that only one had. Here the escape cannot be constructed, so there is no second site to remember.

`repoRelative` can only refuse *more* than your condition did, never less. Pinned by `refuses a path that escapes the root`; a mutation that stops refusing fails on it.

The other seven items are unchanged from the earlier comment.

---
_Generated by [Claude Code](https://claude.ai/code)_
